src/test/shell/bazel/bazel_java_tools_test.sh

```diff
@@ -173,4 +173,89 @@ function test_java_tools_has_proguard() {
   expect_path_in_java_tools "java_tools/third_party/java/proguard/GPL.html"
 }
 
+function test_java_tools_has_java_tools_defs() {
+  expect_path_in_java_tools "java_tools/java_tools_defs.bzl"
+}
+
+function test_java_tools_toolchain_builds() {
+  local java_tools_rlocation=$(rlocation io_bazel/src/java_tools_${JAVA_TOOLS_JAVA_VERSION}.zip)
+  local java_tools_zip_file_url="file://${java_tools_rlocation}"
+  if "$is_windows"; then
+        java_tools_zip_file_url="file:///${java_tools_rlocation}"
+  fi
+  cat >WORKSPACE <<EOF
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+http_archive(
+    name = "local_java_tools",
+    urls = ["${java_tools_zip_file_url}"]
+)
+EOF
+  bazel build @local_java_tools//:toolchain || fail "toolchain failed to build"
+}
+
+function test_java_tools_singlejar_builds() {
+  if "$is_windows"; then
+        # TODO(#8614) singlejar and ijar don't build from source on Windows.
+        echo "Skipping test." && return
+  fi
+  local java_tools_rlocation=$(rlocation io_bazel/src/java_tools_${JAVA_TOOLS_JAVA_VERSION}.zip)
+  local java_tools_zip_file_url="file://${java_tools_rlocation}"
+  if "$is_windows"; then
+        java_tools_zip_file_url="file:///${java_tools_rlocation}"
+  fi
+  cat >WORKSPACE <<EOF
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+http_archive(
+    name = "local_java_tools",
+    urls = ["${java_tools_zip_file_url}"]
+)
+EOF
+  bazel build @local_java_tools//:singlejar_cc_bin || fail "singlejar failed to build"
+}
+
+function test_java_tools_ijar_builds() {
+  if "$is_windows"; then
+        # TODO(#8614) singlejar and ijar don't build from source on Windows.
+        echo "Skipping test." && return
+  fi
+  local java_tools_rlocation=$(rlocation io_bazel/src/java_tools_${JAVA_TOOLS_JAVA_VERSION}.zip)
+  local java_tools_zip_file_url="file://${java_tools_rlocation}"
+  if "$is_windows"; then
+        java_tools_zip_file_url="file:///${java_tools_rlocation}"
+  fi
+  cat >WORKSPACE <<EOF
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+http_archive(
```

Comment:
Will local_repository work instead of http_archive?

Reply:
local_repository only works if we unzip the archive first, which means additional setup and complexity in the tests. My intention was also to test the archive in the same way it is used by Bazel and Java users.